RecoTracker/FinalTrackSelectors/plugins/TrackSelectorByRegion.cc

```diff
@@ -0,0 +1,166 @@
+      desc.add<edm::ParameterSetDescription>("RegionPSet", region_par);
+      descriptions.add("trackSelectorByRegion", desc);
+    }
+
+  private:
+    using MaskCollection = std::vector<bool>;
+
+    void produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup&) const override {
+      // products
+      auto mask = std::make_unique<MaskCollection>();                  // mask w/ the same size of the input collection
+      auto output_tracks = std::make_unique<reco::TrackCollection>();  // selected output collection
+
+      edm::Handle<edm::OwnVector<TrackingRegion>> regionsHandle;
+      iEvent.getByToken(inputTrkRegionToken_, regionsHandle);
+
+      edm::Handle<reco::TrackCollection> tracksHandle;
+      iEvent.getByToken(tracksToken_, tracksHandle);
+
+      //	std::cout << "[TrackSelectorByRegion::produce] tracksHandle.isValid ? " << (tracksHandle.isValid() ? "YEAP" : "NOPE") << std::endl;
+      if (tracksHandle.isValid()) {
+        const auto& tracks = *tracksHandle;
+        mask->assign(tracks.size(), false);
+
+        if (regionsHandle.isValid()) {
+          const auto& regions = *regionsHandle;
+
+          const auto n_regions = regions.size();
+          std::vector<float> etaMin;
+          etaMin.reserve(n_regions);
+          std::vector<float> etaMax;
+          etaMax.reserve(n_regions);
+          std::vector<float> phiMin;
+          phiMin.reserve(n_regions);
+          std::vector<float> phiMax;
+          phiMax.reserve(n_regions);
+          std::vector<float> phi0;
+          phi0.reserve(n_regions);
+          std::vector<float> phi0margin;
+          phi0margin.reserve(n_regions);
+          std::vector<math::XYZPoint> origin;
+          origin.reserve(n_regions);
+          std::vector<float> zBound;
+          zBound.reserve(n_regions);
+
+          for (const auto& tmp : regions)
+            if (const auto* etaPhiRegion = dynamic_cast<const RectangularEtaPhiTrackingRegion*>(&tmp)) {
+              const auto& etaRange = etaPhiRegion->etaRange();
+              const auto& phiMargin = etaPhiRegion->phiMargin();
+
+              etaMin.push_back(etaRange.min());
+              etaMax.push_back(etaRange.max());
+
+              phiMin.push_back(etaPhiRegion->phiDirection() - phiMargin.left());
+              phiMax.push_back(etaPhiRegion->phiDirection() + phiMargin.right());
+              phi0.push_back(etaPhiRegion->phiDirection());
+              phi0margin.push_back(phiMargin.right());
+
+              GlobalPoint gp = etaPhiRegion->origin();
+              origin.push_back(math::XYZPoint(gp.x(), gp.y(), gp.z()));
+              zBound.push_back(etaPhiRegion->originZBound());
+            }
+
+          size_t it = 0;
+          //	    std::cout << "tracks: " << tracks.size() << std::endl;
+          for (auto trk : tracks) {
+            const auto pt = trk.pt();
+            if (pt < minPt_) {
+              //		std::cout << " KO !!! for pt" << std::endl;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant